### PR TITLE
Optimize order date generation

### PR DESCRIPTION
**Pull request by layoutd**
Replaces inefficient date array building with direct timestamp calculation.

Builds on: https://github.com/woocommerce/wc-smooth-generator/pull/182

## Problem
For each order, the old code built an array of all dates in the range (e.g., 664 dates for 2 years), then used array_rand(). For 500 orders over 2 years, this created 332,000 unnecessary date strings.

## Solution
Calculate random offset directly using timestamps instead of building arrays.

## Impact
- Instant date calculation vs building 664-element arrays per order
- Eliminates wasted memory allocation
- Same uniform random distribution
- Orders are created in chronological order (smaller order ID = earlier order)

## Tests

**Test 1: Batch generation with date range produces chronological orders**
```bash
wp wc generate orders 10 --date-start=2024-01-01 --date-end=2024-12-31
```
Expected: Order dates should be in ascending or equal order (no date inversions where later ID has earlier date)

**Test 2: Same-day date range edge case**
```bash
wp wc generate orders 5 --date-start=2024-06-15 --date-end=2024-06-15
```
Expected: All orders created with date 2024-06-15 (different times, randomized)

**Test 3: Error handling for failed order generation**
```bash
# Attempt to create orders when no products exist (should skip failed orders gracefully)
wp wc generate orders 5
```
Expected: Error logged but batch continues; indicates 0 orders generated in the end.

**Test 4: Large multi-year batch maintains chronological order**
```bash
wp wc generate orders 100 --date-start=2023-01-01 --date-end=2024-12-31
```
Expected: Date range covers the specified period with proper distribution

### CODE REVIEW

File: includes/Generator/Order.php [hunk 1]
```diff
@@ -256,14 +280,14 @@ protected static function get_date_created( $assoc_args ) {
...
-		$dates = array();
-		$date  = strtotime( $start );
-		while ( $date <= strtotime( $end ) ) {
-			$dates[] = date( 'Y-m-d', $date );
-			$date    = strtotime( '+1 day', $date );
-		}
+		// Use timestamp-based random selection for single order generation
+		$start_timestamp = strtotime( $start );
+		$end_timestamp   = strtotime( $end );
+		$days_between    = (int) ( ( $end_timestamp - $start_timestamp ) / DAY_IN_SECONDS );
```
[the remark this answers is not in the document]
Comment (layoutd) [reply includes/Generator/Order.php:318]: Addressed in 34f3054601c340dd53b00f6162f8a77eb19dbf4e

File: includes/Generator/Order.php [hunk 2]
```diff
@@ -721,4 +745,42 @@ protected static function create_refund( $order, $force_partial = false, $previo
...
+			$end   = $args['date-end'];
+		} else {
+			// No date range specified, return array of current dates
+			return array_fill( 0, $count, $current );
+		}
+
+		$start_timestamp = strtotime( $start );
+		$end_timestamp   = strtotime( $end );
+		$days_between    = (int) ( ( $end_timestamp - $start_timestamp ) / DAY_IN_SECONDS );
+
```
[the remark this answers is not in the document]
Comment (layoutd) [reply includes/Generator/Order.php:837]: Addressed in 09500f2ab1c0327f70ae97d3735224031d01f350

**Comment by layoutd** [comment]
Related to Test 3, I've added another PR because I don't think "Success: 0 orders generated…" is a very accurate message. ("Task failed successfully"): https://github.com/woocommerce/wc-smooth-generator/pull/189

File: includes/Generator/Order.php [hunk 3]
```diff
@@ -790,4 +819,47 @@ protected static function calculate_refund_date( $order, $previous_refund = null
...
+			$end   = $current;
+		} elseif ( ! empty( $args['date-start'] ) && ! empty( $args['date-end'] ) ) {
+			$start = $args['date-start'];
+			$end   = $args['date-end'];
+		} else {
+			// No date range specified, return array of current dates
+			return array_fill( 0, $count, $current );
+		}
+
+		$start_timestamp = strtotime( $start );
```
Comment (chihsuan) [inline includes/Generator/Order.php:834]: Should we validate that `date-start` and `date-end` are in the expected format and that the start date is earlier than the end date before we use them?  Or it's already done in some other places?

File: includes/Generator/Order.php [hunk 4]
```diff
@@ -236,17 +243,26 @@ public static function batch( $amount, array $args = array() ) {
...
-			if ( ! $order ) {
+			if ( ! $order instanceof \WC_Order ) {
 				error_log( "Batch generation failed: Order {$i} of {$amount} could not be generated" );
 				continue;
 			}
 			$order_ids[] = $order->get_id();
 		}
 
+		// Clear batch dates after generation
+		self::$batch_dates = null;
```
Comment (chihsuan) [inline includes/Generator/Order.php]: Instead of using a static variable to store the batch dates, what do you think about just passing the batch dates as an argument to `generate` method? This way, we don't need to worry about shared state between different instances of the class. We can add a new parameter, or modify `$assoc_args` to include the batch dates.

**Review by chihsuan — COMMENTED** [review]
Great optimization! 💯 I was able to test it successfully according to the instructions. I just left a couple of minor suggestions.

File: includes/Generator/Order.php (hunk 3 above)
Comment (layoutd) [reply includes/Generator/Order.php:834]: It's actually not validated anywhere, but I'll add another PR later to validate the params.

File: includes/Generator/Order.php (hunk 4 above)
Comment (layoutd) [reply includes/Generator/Order.php]: Yeah, that probably makes more sense: https://github.com/woocommerce/wc-smooth-generator/pull/185/commits/40d6ad67668c0337c198685130d76517de4fa534
